client/homebrew/utils/versionHistory.js

```diff
@@ -43,74 +50,110 @@ function updateStoredBrew(brew, slot = 0) {
 		expireAt : new Date()
 	};
 
+	if(global.config?.history?.HISTORY_SAVE_DELAYS){
+		HISTORY_SAVE_DELAYS = global.config?.history?.HISTORY_SAVE_DELAYS;
+	}
+
 	archiveBrew.expireAt.setMinutes(archiveBrew.expireAt.getMinutes() + HISTORY_SAVE_DELAYS[slot]);
 
 	const key = getKeyBySlot(brew, slot);
-	localStorage.setItem(key, JSON.stringify(archiveBrew));
+
+	return [key, archiveBrew];
 }
 
 
-export function historyExists(brew){
-	return Object.keys(localStorage)
-		.some((key)=>{
-			return key.startsWith(`${HISTORY_PREFIX}-${brew.shareId}`);
-		});
+async function createHBStore(){
+	return await IDB.createStore(HB_DB, HB_STORE);
+}
+
+export async function historyCheck(brew){
+	if(!IDB) return false;
+	const historyExists = await IDB.keys(await createHBStore())
+		.then((keys)=>{
+			return [...keys].some((key)=>{
+				return key.startsWith(`${HISTORY_PREFIX}-${brew.shareId}`);
+			});
+		})
+		.catch(()=>{return false;});
+
+	return historyExists;
```

Comment:
```suggestion
	const keys = await IDB.keys(await createHBStore())
		.catch(() => []);
	return keys.some(key => key.startsWith(`${HISTORY_PREFIX}-${brew.shareId}`));

```

Reply:
This is implemented in the latest commit, with a minor tweak or two to get it working properly.